src/libpmem2/badblocks_ndctl.c

```diff
@@ -696,24 +696,45 @@ badblocks_clear(const char *file, struct badblocks *bbs)
 
 	ASSERTne(bbs, NULL);
 
-	os_stat_t st;
-	if (os_stat(file, &st) < 0) {
-		ERR("!stat %s", file);
+	struct pmem2_source src;
```

Comment:
This function shouldn't be used for anything - all callers should instead call the external pmem2 API.

Reply:
What function? badblocks_clear()? - removing the old API will be the next step, probably in the next PR.